algorithms/java/src/org/openda/algorithms/kalmanFilter/SteadyStateFilter.java

```diff
@@ -208,24 +213,86 @@ public void analysis(IStochObserver observations, IVector obsValues, IVector pre
 
 		int m=innovation.getSize();
 		IVector pred_a = this.mainModel.getObservationOperator().getObservedValues(observations.getObservationDescriptions());
-		for(int i=0;i<m;i++){
+		ArrayList<Integer> missingObservationIndices = new ArrayList<>();
+		ArrayList<Integer> availableObservationIndices = new ArrayList<>();
+		for (int i = 0; i < gainStorageObservationIdsArray.length; i++) {
```

Comment:
You mentioned that documentation will be part of the repository. I suggest to briefly explain what is going on + a link to the detailed description  in the document.